x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/multi_metric_view/metric_selection.tsx

```diff
@@ -0,0 +1,196 @@
+  }
+
+  // subscribe to progress
+  useEffect(() => {
+    jobCreator.subscribeToProgress(setProgress);
+  }, []);
+
+  // subscribe to results
+  useEffect(() => {
+    resultsLoader.subscribeToResults(setResultsWrapper);
+  }, []);
+
+  // watch for changes in detector list length
+  useEffect(() => {
+    jobCreator.removeAllDetectors();
+    aggFieldPairList.forEach(pair => {
+      jobCreator.addDetector(pair.agg, pair.field);
+    });
+    jobCreatorUpdate();
+    loadCharts();
+    setIsValid(aggFieldPairList.length > 0);
+  }, [aggFieldPairList.length]);
+
+  // watch for change in jobCreator
+  useEffect(() => {
+    if (jobCreator.start !== start || jobCreator.end !== end) {
+      setStart(jobCreator.start);
+      setEnd(jobCreator.end);
+      loadCharts();
+    }
+    setSplitField(jobCreator.splitField);
+  }, [jobCreatorUpdated]);
+
+  // watch for changes in split field.
+  // load example field values
+  // changes to fieldValues here will trigger the card effect
+  useEffect(() => {
+    if (splitField !== null) {
+      chartLoader
+        .loadFieldExampleValues(splitField)
+        .then(setFieldValues)
+        .catch(() => {});
+    } else {
+      setFieldValues([]);
+    }
+  }, [splitField]);
+
+  // watch for changes in the split field values
+  // reload the charts
+  useEffect(() => {
+    loadCharts();
+  }, [fieldValues]);
+
+  function getChartSettings(): ChartSettings {
+    const cs = {
+      ...defaultChartSettings,
+      intervalMs: chartInterval.getInterval().asMilliseconds(),
+    };
+    if (aggFieldPairList.length > 2) {
+      cs.cols = 3;
+      cs.height = '150px';
+      cs.intervalMs = cs.intervalMs * 3;
+    } else if (aggFieldPairList.length > 1) {
+      cs.cols = 2;
+      cs.height = '200px';
+      cs.intervalMs = cs.intervalMs * 2;
+    }
+    return cs;
+  }
+
+  async function loadCharts() {
+    const cs = getChartSettings();
+    setChartSettings(cs);
+
+    if (aggFieldPairList.length > 0) {
+      const resp: LineChartData = await chartLoader.loadLineCharts(
+        jobCreator.start,
+        jobCreator.end,
+        aggFieldPairList,
+        jobCreator.splitField,
+        fieldValues.length > 0 ? fieldValues[0] : null,
+        cs.intervalMs
+      );
+
+      setLineChartsData(resp);
+    }
+  }
+
+  return (
+    <Fragment>
+      {lineChartsData && (
+        <ChartGrid
+          aggFieldPairList={aggFieldPairList}
+          chartSettings={chartSettings}
+          splitField={splitField}
+          fieldValues={fieldValues}
+          lineChartsData={lineChartsData}
+          modelData={modelData}
+          anomalyData={anomalyData}
+          deleteDetector={isActive ? deleteDetector : undefined}
+          jobType={jobCreator.type}
+        />
+      )}
+      {isActive && (
+        <MetricSelector
+          fields={fields}
+          detectorChangeHandler={detectorChangeHandler}
+          selectedOptions={selectedOptions}
+          maxWidth={560}
```

Comment:
nit: could this become a common constant? it's hard coded twice in two different files.